processmanager/manager.go

```diff
@@ -267,6 +267,21 @@ func (pm *ProcessManager) ConvertTrace(trace *host.Trace) (newTrace *libpf.Trace
 				}
 			}
 
+			if pm.eim.IsGolang(frame.File) {
```

Comment:
This is fairly ugly way to plug this in. Makes me wonder if the integration should be done in some other way. Perhaps split the interpreter into unwinder / symbolization stages. Or have some other hooking mechanism to tag the frames that should get Go symbolized. Perhaps add a mechanism to specify the ebpf sent frame type on per-DSO or mapping range basis.

Reply:
Happy to discuss alternatives. For the moment, I think, this is the least intrusive way. This current approach should also be easily adaptable if the interpreter interface differentiates at some point between unwinding and symbolization.

Reply:
To clarify, I really don't like the special golang check here. As minimum, I'd completely remove the `IsGolang` call above (And introduction if it in `eim`). We can just directly call `symbolizeFrame` here. And make sure the existing interpreters ignore native frame (should be the case).

Or do you see some practical reasons why this check should be here?

Reply:
Your suggestion sounds reasonable to me. It just needs some logic on the eBPF side to report Go frames for the respective executables. And I do think introducing a dedicated Go unwinder in eBPF doesn't make sense but the native unwinder needs to be extended in some way to switch the frame type at some point.

Reply:
Applied the suggestion with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/408/commits/7128306e545b20ae6e7364845971be884dc6c623